clang/tools/clang-linker-wrapper/LinkerWrapperOpts.td

```diff
@@ -150,6 +150,9 @@ def sycl_device_lib_EQ : CommaJoined<["--", "-"], "sycl-device-libraries=">,
 def sycl_device_library_location_EQ : Joined<["--", "-"],
   "sycl-device-library-location=">, Flags<[WrapperOnlyOption]>,
   HelpText<"Location of SYCL device library files">;
+def sycl_nvptx_device_lib_EQ : CommaJoined<["--", "-"], "sycl-nvptx-device-libraries=">,
```

Comment:
Note sure we should have `nvptx`-specific options if we can avoid it. It makes it harder to update if anything changes in the future with supported SYCL targets, or with how we compile AMD.

Would it be at all possible to link this option up with `-Xsycl-target-backend`? Something like `-Xsycl-target-backend -sycl-device-libraries=...`? Then if there were multiple targets you could specify `-Xsycl-target-backend=nvptx64-nvidia-cuda -sycl-device-libraries=... -Xsycl-target-backend=amdgcn-amd-amdhsa -sycl-device-libraries=...`?

Reply:
Hi @frasercrmck 

Thanks so much for taking a look. One of the caveats in this PR is that this is not expected to be the final version to be upstreamed. This is expected to be an intermediate solution that is expected to functionally work. One of the key steps we intend t perform before upstreaming will be to refactor how we have introduced non-generic clang-linker-wrapper options. About this particular option, I feel that the we might be able to replace this option completety.

Hope it's agreeable from this PR's perspective.

Sincerely

Reply:
Ah okay, I wasn't aware of that. That's fine, then. I just worry about the impact of changing flags once there's code out there that might be relying on them.